libraries/bezier-rs/src/subpath/transform.rs

```diff
@@ -78,13 +89,190 @@ impl<ManipulatorGroupId: crate::Identifier> Subpath<ManipulatorGroupId> {
+				in_handle: group.out_handle,
+				out_handle: group.in_handle,
+				id: ManipulatorGroupId::new(),
+			})
+			.collect::<Vec<ManipulatorGroup<ManipulatorGroupId>>>()
+	}
+
+	/// Returns a [Subpath] with a reversed winding order.
+	pub fn reverse(&self) -> Subpath<ManipulatorGroupId> {
+		Subpath {
+			manipulator_groups: Subpath::reverse_manipulator_groups(&self.manipulator_groups),
+			closed: self.closed,
+		}
+	}
+
+	/// Returns an open [Subpath] that results from trimming the original Subpath between the points corresponding to `t1` and `t2`.
+	/// If `t1` > `t2`, then behavior for computing the resulting Subpath will differ depending on whether the original Subpath is open or closed:
+	/// - If the original Subpath was open, an open subpath from `t2` to `t1` will be returned (with winding in the same direction as the original Subpath).
+	/// - If the original Subpath was closed, an open subpath from `t1` to `t2` that crosses the break between 1 and 0 will be returned (winding in the same direction as the original Subpath from `t1` to `t = 1 = 0` to `t2`).
+	///
+	/// If a path winding in the reverse direction is desired, call `trim` on the `Subpath` returned from `Subpath::reverse`.
+	/// <iframe frameBorder="0" width="100%" height="400px" src="https://graphite.rs/bezier-rs-demos#subpath/trim/solo" title="Trim Demo"></iframe>
+	pub fn trim(&self, t1: SubpathTValue, t2: SubpathTValue) -> Subpath<ManipulatorGroupId> {
+		// Return a clone of the Subpath if it is not long enough to be a valid Bezier
+		if self.manipulator_groups.is_empty() {
+			return Subpath {
+				manipulator_groups: vec![],
+				closed: self.closed,
+			};
+		}
+
+		let (mut t1_curve_index, mut t1_curve_t) = self.t_value_to_parametric(t1);
+		let (mut t2_curve_index, mut t2_curve_t) = self.t_value_to_parametric(t2);
+
+		// The only case where t would be 1 is when the input parameter refers to the the very last point on the subpath.
+		// We want these index and t pairs to always represent that point as the next curve index with t == 0.
+		if t1_curve_t == 1. {
+			t1_curve_index += 1;
+			t1_curve_t = 0.;
+		}
+		if t2_curve_t == 1. {
+			t2_curve_index += 1;
+			t2_curve_t = 0.;
+		}
+
+		// Check if the trimmed result is in the reverse direction
+		let are_arguments_reversed = t1_curve_index > t2_curve_index || (t1_curve_index == t2_curve_index && t1_curve_t > t2_curve_t);
+		if !self.closed && are_arguments_reversed {
+			(t1_curve_index, t2_curve_index) = (t2_curve_index, t1_curve_index);
+			(t1_curve_t, t2_curve_t) = (t2_curve_t, t1_curve_t);
+		}
+
+		// Get a new list from the manipulator groups that will be trimmed at the ends to form the resulting subpath.
+		// The list will contain enough manipulator groups such that the later code simply needs to trim the first and last bezier segments
+		// and then update the values of the corresponding first and last manipulator groups accordingly.
+		let mut cloned_manipulator_groups = self.manipulator_groups.clone();
```

Comment:
Hmm, does this need to include a call to `ManipulatorGroupId::new()` for each of these cloned manipulator groups?

Reply:
No, it doesn't need to regenerate the id, as the id only has to be unique in the subpath.

Depending on how the vector nodes work, we may also want to avoiding regenerating the id of manipulator groups that are unchanged by the splitting operation (for example in `reverse_manipulator_groups` above).

This would allow the node network `Move Manipulator Points By Id --> Split Subpath` to be rearranged into `Split Subpath --> Move Manipulator Points By Id` which would give the same results when the moving and splitting modified different manipulator groups. I'm unsure if this is needed though @Keavon and @otdavies?

If we do not care about the consistency of ids, then we can regenerate them as part of the clone implementation on the `ManipulatorGroupId` type.

Reply:
Based on the discussion on discord, I'm assuming we do not need to regenerate these ids and will merge the PR as it is now.